apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/NovaComputeServiceAdapter.java

```diff
@@ -123,41 +128,32 @@ public NovaComputeServiceAdapter(NovaApi novaApi, @Region Supplier<Set<String>>
          options.networks(templateOptions.getNetworks());
       }
 
-      Optional<String> privateKey = Optional.absent();
       if (templateOptions.getKeyPairName() != null) {
          options.keyPairName(templateOptions.getKeyPairName());
-         KeyPair keyPair = keyPairCache.getIfPresent(RegionAndName.fromRegionAndName(template.getLocation().getId(), templateOptions.getKeyPairName()));
-         if (keyPair != null && keyPair.getPrivateKey() != null) {
-            privateKey = Optional.of(keyPair.getPrivateKey());
-            credentialsBuilder.privateKey(privateKey.get());
-         }
```

Comment:
We need this code. Users using an existing, pre-created keypair won't be able to login to the node since the private key won't be populated to the login credentials.